gcloud-java-storage/src/main/java/com/google/gcloud/spi/StorageRpc.java

```diff
@@ -106,9 +108,12 @@ public Y y() {
     public BatchRequest(Iterable<Tuple<StorageObject, Map<Option, ?>>> toDelete,
         Iterable<Tuple<StorageObject, Map<Option, ?>>> toUpdate,
         Iterable<Tuple<StorageObject, Map<Option, ?>>> toGet) {
-      this.toDelete = ImmutableList.copyOf(toDelete);
-      this.toUpdate = ImmutableList.copyOf(toUpdate);
-      this.toGet = ImmutableList.copyOf(toGet);
+      this.toDelete = ImmutableList.copyOf(
```

Comment:
See my comment bellow regarding using the Batch on the service layer rather the RPC layer.